### ISSUE TITLE
signin ready event not triggered after forgot password is clicked

### ISSUE THREAD

**Issue by matiya** [auth0/lock#2005]
### Describe the problem

I have attached simple custom logic to the lock.on("signin ready") event. It works as expected when the sign in box is displayed. 
However, when the user clicks the "forgot password" link and then presses the back arrow to return to the sign in box, this event is not triggered. 

### What was the expected behavior?

The sign in ready event should be triggered when the sign in is ready, independent of the previous state.

### Reproduction

- Open the sign in box
- Custom logic is executed
- Press "forgot password" link
- Press the back arrow to return to the sign in box
- Custom logic is not executed

### Environment
- Lock 11.30

**Comment by stevehobbsdev** [comment]
Hey @matiya, as a first step I tried to reproduce this in the built-in playground (clone the repo and run `yarn start`), which automatically logs events to the UI for debugging. Following the steps you provide, I can see the`signin ready` event being fired twice; once on load and again when moving back to the login screen. Can you confirm that you're seeing the same?


https://user-images.githubusercontent.com/766403/121341299-6f61a380-c918-11eb-80c2-2537a689c982.mov

**Comment by matiya** [comment]
Hi @stevehobbsdev, thank you for the reply. It turns out that I was testing this wrong. 

I replaced my custom function that was being called on every "signin ready" event with a simple alert() and in both scenarios I could see the alert.

Now I realize that my code is failing because the event is removing (correctly) custom HTML nodes that I have attached to the HTML nodes that lock draws. 

Sorry for reporting a bug that isn't a bug.
